src/uct/tcp/tcp_ep.c

```diff
@@ -936,8 +936,6 @@ ucs_status_t uct_tcp_ep_am_short(uct_ep_h uct_ep, uint8_t am_id, uint64_t header
                 uct_tcp_ep_mod_events(ep, UCS_EVENT_SET_EVWRITE, 0);
                 return UCS_OK;
             }
-
```

Comment:
what about case when ```status == UCS_ERR_NO_PROGRESS```?

Reply:
yes, right
this was handled in `uct_tcp_ep_ctx_buf_need_progress()`, because if `NO_PROGRESS` sent length `=` `0`, i.e. it will return `UCS_OK`

Reply:
this relation is not really obvious (and can be easily broken later).
Please add `assert (status==UCS_OK)` instead of removed line then

Reply:
done